src/ansible_navigator/image_manager/puller.py

```diff
@@ -128,8 +132,14 @@ def prologue_stdout(self):
         """print a little value added information"""
         messages = [("Execution environment image name:", self._image)]
         messages.append(("Execution environment image tag:", self._image_tag))
+        if isinstance(self._arguments, list):
```

Comment:
Missed this the first time... The type above (in the constructor) says this __has__ to be a `List[str]`, so this `isinstance` should always be true, unless I'm missing something.

Reply:
That's concerning to me. If the type was wrong, then why is it type-checking? Is this a result of #956 not being merged yet?

Reply:
No, even with that one in now, mypy can't track the type.  but I think we talked about this and have an idea of where to go with SettingsEntry....

Reply:
With above suggested change this check is not required?

Reply:
Yeah, the type of each settings parameter is lost along the way:

```
def pull_image(args: ApplicationConfiguration):
    """pull the image if required"""
    reveal_type(args.pull_arguments)
    image_puller = ImagePuller(
        container_engine=args.container_engine,
        image=args.execution_environment_image,
        arguments=args.pull_arguments,
        pull_policy=args.pull_policy,
    )
```
```
(venv) x1 ➜  ansible-navigator git:(podman_pull) ✗ mypy
src/ansible_navigator/cli.py: note: In function "pull_image":
src/ansible_navigator/cli.py:49:17: note: Revealed type is "Any"
```

Rick is working on a refactor of the settings so the type information is preserved...